benchmark/bench_fused_moe.py

```diff
@@ -0,0 +1,306 @@
+    x1 = F.silu(x1)
+    x3 = torch.einsum("ti, taoi -> tao", x, w3_weights)
+    expert_outs = torch.einsum("tao, taio -> tai", (x1 * x3), w2_weights)
+    return torch.einsum("tai,ta -> ti", expert_outs, topk_weights.to(expert_outs.dtype))
+
+
+def fused_moe_torch_compile(
+    x,
+    w1,
+    w2,
+    input_gating,
+    topk,
+    use_fp8_w8a8=False,
+    w1_scale=None,
+    w2_scale=None,
+    a1_scale=None,
+    a2_scale=None,
+):
+    return fused_moe_torch(
+        x,
+        w1,
+        w2,
+        input_gating,
+        topk,
+    )
+
+
+def fused_moe_sglang_api(
+    x,
+    w1,
+    w2,
+    input_gating,
+    topk,
+):
+    num_tokens = x.shape[0]
+    topk_weights = torch.empty(num_tokens, topk, dtype=torch.float32, device=x.device)
+    topk_indices = torch.empty(num_tokens, topk, dtype=torch.int32, device=x.device)
+
+    topk_softmax(
+        topk_weights,
+        topk_indices,
+        input_gating,
+        renormalize=True,
+    )
+    return fused_experts(
+        x,
+        w1,
+        w2,
+        topk_weights,
+        topk_indices,
+    )
+
+
+@triton.testing.perf_report(
+    triton.testing.Benchmark(
+        x_names=[
+            "num_tokens",
+            "num_experts",
+            "topk",
+            "hidden_size",
+            "shard_intermediate_size",
+            "dtype",
+            "block_shape",
+        ],
+        x_vals=configs,
+        line_arg="provider",
+        line_vals=[
+            "torch_compile",
+            "sgl_kernel",
+        ],
+        line_names=[
+            "torch_compile",
+            "sgl_kernel",
+        ],
+        styles=[
+            ("blue", "-"),
+            ("green", "-"),
+        ],
+        ylabel="Time (ms)",
+        plot_name="fused-moe-performance",
+        args={},
+    )
+)
+def benchmark(
+    num_tokens,
+    num_experts,
+    topk,
+    hidden_size,
+    shard_intermediate_size,
+    dtype,
+    block_shape,
+    provider,
+):
+    print(
+        f"benchmark {provider} with batch_size={num_tokens} hidden_size={hidden_size} shard_intermediate_size={shard_intermediate_size}"
+    )
+    torch.set_default_device("xpu")
+    torch.xpu.manual_seed_all(0)
+
+    x = torch.randn(num_tokens, hidden_size, dtype=dtype)
+
+    w1 = torch.randn(num_experts, shard_intermediate_size, hidden_size, dtype=dtype)
+    w2 = torch.randn(
+        num_experts, hidden_size, shard_intermediate_size // 2, dtype=dtype
+    )
+
+    input_gating = torch.randn(num_tokens, num_experts, dtype=dtype)
+
+    if provider == "torch_compile":
+        api_func = fused_moe_torch_compile
+        api_kwargs = {
+            "x": x,
+            "w1": w1,
+            "w2": w2,
+            "input_gating": input_gating,
+            "topk": topk,
+        }
+    else:
+        api_func = fused_moe_sglang_api
+        api_kwargs = {
+            "x": x,
+            "w1": w1,
+            "w2": w2,
+            "input_gating": input_gating,
+            "topk": topk,
+        }
```

Comment:
```suggestion
if provider == "torch_compile":
    api_func = fused_moe_torch_compile
else:
    api_func = fused_moe_sglang_api

api_kwargs = {
    "x": x,
    "w1": w1,
    "w2": w2,
    "input_gating": input_gating,
    "topk": topk,
}
```

remove the duplicate.